src/main/scala/firrtl/transforms/InferResets.scala

```diff
@@ -163,6 +165,12 @@ class InferResets extends Transform {
               val values = conLookup(key).getOrElse(Nil) ++ altLookup(key).getOrElse(Nil)
               map(key) = values
             }
+          case lhs@ DefNode(_, _, rhs) => rhs match {
+            case Mux(_, tval, fval, _) =>
+              markResetDriver(WRef(lhs), tval)
+              markResetDriver(WRef(lhs), fval)
+            case _ => markResetDriver(WRef(lhs), rhs)
+          }
```

Comment:
Not having this was probably just an oversight since one could express this in Chisel so it wouldn't work properly with the current transform.

I'll note that moving this after ExpandConnects and ExpandWhens means we can vastly simplify much of the above logic.